source/vulkan/vulkan_impl_command_list.cpp

```diff
@@ -864,43 +865,169 @@ void reshade::vulkan::command_list_impl::resolve_texture_region(api::resource sr
+		blit_info.dstOffsets[1] = VkOffset3D { int32_t(dst_extent.width), int32_t(dst_extent.height), 1 };
+		blit_info.srcOffsets[0] = VkOffset3D { 0, 0, 0 };
+		blit_info.srcOffsets[1] = VkOffset3D { int32_t(src_extent.width), int32_t(src_extent.height), 1 };
+
+		const VkExtent3D src_copy_extent = {
+			uint32_t(blit_info.srcOffsets[1].x - blit_info.srcOffsets[0].x),
+			uint32_t(blit_info.srcOffsets[1].y - blit_info.srcOffsets[0].y),
+			uint32_t(blit_info.srcOffsets[1].z - blit_info.srcOffsets[0].z) };
+
+		const VkExtent3D dst_copy_extent = {
+			uint32_t(blit_info.dstOffsets[1].x - blit_info.dstOffsets[0].x),
+			uint32_t(blit_info.dstOffsets[1].y - blit_info.dstOffsets[0].y),
+			uint32_t(blit_info.dstOffsets[1].z - blit_info.dstOffsets[0].z) };
+
+		VkImageResolve resolve_region;
+		resolve_region.srcSubresource = blit_info.srcSubresource;
+		resolve_region.srcOffset = blit_info.srcOffsets[0];
+		resolve_region.dstSubresource = blit_info.srcSubresource;
+		resolve_region.dstOffset = blit_info.srcOffsets[0];
+		resolve_region.extent = src_copy_extent;
+
+		VkImageSubresourceRange dst_subresource_range;
+		dst_subresource_range.aspectMask = resolve_region.dstSubresource.aspectMask;
+		dst_subresource_range.baseMipLevel = resolve_region.dstSubresource.mipLevel;
+		dst_subresource_range.levelCount = 1;
+		dst_subresource_range.baseArrayLayer = resolve_region.dstSubresource.baseArrayLayer;
+		dst_subresource_range.layerCount = resolve_region.dstSubresource.layerCount;
+
+		VkImageViewUsageCreateInfo usage_info = { VK_STRUCTURE_TYPE_IMAGE_VIEW_USAGE_CREATE_INFO };
+		usage_info.usage = VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT;
+
+		VkImageViewCreateInfo img_create_info = { VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO, &usage_info };
+		img_create_info.image = dst.handle;
+		img_create_info.viewType = VK_IMAGE_VIEW_TYPE_2D_ARRAY;
+		img_create_info.format = src_data->create_info.format;
+		img_create_info.subresourceRange = dst_subresource_range;
+
+		VkImageView dst_img_view;
+		if (vk.CreateImageView(_device_impl->_orig, &img_create_info, nullptr, &dst_img_view)) {
+			assert(false);
+			return;
+		}
 
-	region.dstOffset = { dst_x, dst_y, dst_z };
+		VkImageSubresourceRange src_subresource_range;
+		src_subresource_range.aspectMask = resolve_region.srcSubresource.aspectMask;
+		src_subresource_range.baseMipLevel = resolve_region.srcSubresource.mipLevel;
+		src_subresource_range.levelCount = 1;
+		src_subresource_range.baseArrayLayer = resolve_region.srcSubresource.baseArrayLayer;
+		src_subresource_range.layerCount = resolve_region.srcSubresource.layerCount;
 
-	vk.CmdResolveImage(_orig,
-		(VkImage)src.handle, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
-		(VkImage)dst.handle, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
-		1, &region);
+		img_create_info.image = src.handle;
+		img_create_info.subresourceRange = src_subresource_range;
+
+		VkImageView src_img_view;
+		if (vk.CreateImageView(_device_impl->_orig, &img_create_info, nullptr, &src_img_view)) {
```

Comment:
This is a big nono and a huge memory leak. These resources are never freed, so right now every time this is called (every frame) additional memory is allocated and never cleaned up again.

ReShade does already create a default image view for depth-stencil images (with a subresource range that encompasses the entire image) however, which is fortunate, as that can be used in the case where no region is passed to `resolve_texture_region`. Support for region resolve would have to be dropped for now though.

You can get to the default views via `src_data->default_view` and `dst_data->default_view`.